pkg/gather/gather.go

```diff
@@ -56,12 +55,15 @@ type ArchiveMetadata struct {
 func CreateAllGatherers(
 	gatherKubeConfig, gatherProtoKubeConfig, metricsGatherKubeConfig *rest.Config,
 	anonymizer *anonymization.Anonymizer, controller *config.Controller,
+	configurator configobserver.Configurator,
 ) []gatherers.Interface {
 	clusterConfigGatherer := clusterconfig.New(
 		gatherKubeConfig, gatherProtoKubeConfig, metricsGatherKubeConfig, anonymizer, controller.Interval,
 	)
 	workloadsGatherer := workloads.New(gatherProtoKubeConfig)
-	conditionalGatherer := conditional.New(gatherProtoKubeConfig, metricsGatherKubeConfig)
+	conditionalGatherer := conditional.New(
+		gatherProtoKubeConfig, metricsGatherKubeConfig, configurator.Config().ConditionalGathererRulesEndpoint,
+	)
```

Comment:
I don't think there's a need to pass the `configobserver.Configurator` instance to the `CreateAllGatherers` function. You already have the `controller` here so you can call `controller.ConditionalGathererRulesEndpoint`.

Reply:
fixed